TAO/orbsvcs/orbsvcs/Event/ECG_Mcast_EH.cpp

```diff
@@ -197,6 +200,7 @@ int
 TAO_ECG_Mcast_EH::delete_unwanted_subscriptions (
     Address_Set& multicast_addresses)
 {
+  ACE_GUARD_RETURN (TAO_SYNCH_MUTEX, ace_mon, this->lock_, -1);
```

Comment:
Thanks for your ideas. I think I'll close this PR and make a new based on your proposal. And I need to make a separate PR for TTY changes that slipped in.